plugins/render_fortran_examples.py

```diff
@@ -0,0 +1,252 @@
+        uptodate = {"c": self.site.GLOBAL_CONTEXT}
+
+        for k, v in self.site.GLOBAL_CONTEXT["template_hooks"].items():
+            uptodate["||template_hooks|{0}||".format(k)] = v.calculate_deps()
+
+        for k in self.site._GLOBAL_CONTEXT_TRANSLATABLE:
+            uptodate[k] = self.site.GLOBAL_CONTEXT[k](kw["default_lang"])
+
+        # save navigation links as dependencies
+        uptodate["nav_links"] = uptodate["c"]["navigation_links"](kw["default_lang"])
+
+        uptodate["kw"] = kw
+
+        examples_template_deps = self.site.template_system.template_deps(
+            "examples.tmpl"
+        )
+
+        index_template_deps = self.site.template_system.template_deps(
+            "fortran-example-index.tmpl"
+        )
+        folder = Path(self.input_folder).resolve()
+        fortran_examples = list(chain(
+            folder.glob("*.f90"),
+            folder.glob("../f77/*.f"),  # hack to combine all Fortran samples on one page
+            folder.glob("../f77/*.cpp"),
+        ))
+        fortran_headers = {
+            "examples": {"name": "Examples", "files": [], "summaries": {}}
+        }
+
+        uptodate["d"] = fortran_headers.keys()
+        uptodate["f"] = list(map(str, fortran_examples))
+
+        for fortran_ex_file in fortran_examples:
+            # Try to get the first comment block in the file to use as a description
+            # Combination of detection for F77, F90, and C/C++ comments
+            fortran_headers["examples"]["files"].append(fortran_ex_file)
+            doc = []
+            in_block_comment = False
+            for line in fortran_ex_file.read_text(encoding="utf-8").split("\n"):
+                line = line.strip()
+                if '*/' in line:
+                    in_block_comment = False
+                    line = line[:line.find('*/')]
+                if line.startswith('/*'):
+                    doc.append(line.lstrip('/* !'))
+                    in_block_comment = True
+                if in_block_comment or line.startswith('!') or line.startswith("//")  or line.startswith('* '):
+                    doc.append(line.lstrip('/* !'))
+                elif line.startswith('c     '):
+                    doc.append(line[6:])
+                elif doc:
+                    break
+            doc = ' '.join(doc).strip()
+
+            print(fortran_ex_file, '======\n', doc, '\n================\n')
+            if not doc:
+                self.logger.warn(
+                    "The Fortran example {!s} doesn't have an appropriate summary. The "
+                    "first comment line of the Fortran file is taken as the "
+                    "summary.".format(fortran_ex_file)
+                )
+            name = fortran_ex_file.stem.replace("_", " ")
+            fortran_headers["examples"]["summaries"][fortran_ex_file.name] = doc
+            out_name = kw["output_folder"].joinpath(
+                self.examples_folder, fortran_ex_file.name + '.html'
```

Comment:
Same here about `Pathlib.with_suffix()`

Reply:
I don't think that's what we want in this case - we're trying to generate a name like `isentropic.f.html` - we don't want to replace the existing `.f` suffix.